composeApp/src/main/kotlin/io/github/couchtracker/ui/components/TimezonePickerDialog.kt

```diff
@@ -0,0 +1,129 @@
+package io.github.couchtracker.ui.components
+
+import androidx.compose.foundation.ExperimentalFoundationApi
+import androidx.compose.foundation.background
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.BoxWithConstraints
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.items
+import androidx.compose.foundation.lazy.rememberLazyListState
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.TravelExplore
+import androidx.compose.material3.AlertDialog
+import androidx.compose.material3.AlertDialogDefaults
+import androidx.compose.material3.HorizontalDivider
+import androidx.compose.material3.Icon
+import androidx.compose.material3.ListItem
+import androidx.compose.material3.ListItemDefaults
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.material3.TextButton
+import androidx.compose.material3.darkColorScheme
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.unit.dp
+import io.github.couchtracker.R
+import io.github.couchtracker.utils.str
+import kotlinx.datetime.Clock
+import kotlinx.datetime.Instant
+import kotlinx.datetime.TimeZone
+import kotlinx.datetime.offsetAt
+
+@Composable
+fun TimezonePickerDialog(
+    timezone: TimeZone?,
+    onTimezoneSelected: (TimeZone) -> Unit,
+    close: () -> Unit,
+) {
+    val instant = remember { Clock.System.now() }
```

Comment:
Issue: this is wrong. When selecting dates in the past this might be off if current and past DST mismatch.

Alternatives:
1. pass a (nullable) `LocalDateTime` to use to calculate the offset, and use the current time only as a fallback
2. do not group this way, but split each timezone on `/` and group based on the first part. Timezones with no `/` can go in an `Other` group. Each group can be sorted alphabetically.
    - this will also make searching a timezone faster, as people will more than likely want to look for their city and not search it by offset
    - this is also similar to how `dpkg-reconfigure tzdata` presents the choice: https://i.sstatic.net/hdYIX.png
    - if we group it like above, the search bar is not even needed

I think alternative no 2 i nicer.